addons/account_edi/models/account_move.py

```diff
@@ -135,6 +136,169 @@ def _compute_edi_show_abandon_cancel_button(self):
+
+        seen_invoice_lines = set()
+        seen_grouped_base_lines = defaultdict(set)
+        for tax_values in invoice_lines_tax_values_list:
+            key = grouping_key(tax_values)
+            serialized_key = _serialize_python_dictionary(key)
+
+            invoice_line = tax_values['base_line_id']
+
+            # Add to invoice global tax amounts.
+            invoice_global_tax_details = res
+            if invoice_line not in seen_invoice_lines:
+                seen_invoice_lines.add(invoice_line)
+                invoice_global_tax_details['base_amount'] += invoice_line.balance
+                invoice_global_tax_details['base_amount_currency'] += invoice_line.amount_currency
+            invoice_global_tax_details['tax_amount'] += tax_values['tax_amount']
+            invoice_global_tax_details['tax_amount_currency'] += tax_values['tax_amount_currency']
+
+            # Add to invoice global tax details.
+            invoice_tax_details = invoice_global_tax_details['tax_details'][serialized_key]
+            if not seen_grouped_base_lines[serialized_key]:
+                invoice_tax_details.update(key)
+            if invoice_line not in seen_grouped_base_lines[serialized_key]:
+                seen_grouped_base_lines[serialized_key].add(invoice_line)
+                invoice_tax_details['base_amount'] += tax_values['base_amount']
+                invoice_tax_details['base_amount_currency'] += tax_values['base_amount_currency']
+            invoice_tax_details['tax_amount'] += tax_values['tax_amount']
+            invoice_tax_details['tax_amount_currency'] += tax_values['tax_amount_currency']
+            invoice_tax_details['records'].append(tax_values)
+
+            # Add to invoice line global tax amounts.
+            invoice_line_global_tax_details = res['invoice_line_tax_details'][invoice_line]
+            invoice_line_global_tax_details['tax_amount'] += tax_values['tax_amount']
+            invoice_line_global_tax_details['tax_amount_currency'] += tax_values['tax_amount_currency']
+
+            # Add to invoice line global tax details.
+            invoice_line_tax_details = invoice_line_global_tax_details['tax_details'][serialized_key]
+            invoice_line_tax_details.update(key)
+            base_operator = max if invoice_line.balance > 0 else min
+            invoice_line_tax_details['base_amount'] = base_operator(
+                invoice_line_tax_details['base_amount'],
+                tax_values['base_amount'],
+            )
+            invoice_line_tax_details['base_amount_currency'] = base_operator(
+                invoice_line_tax_details['base_amount_currency'],
+                tax_values['base_amount_currency'],
+            )
+            invoice_line_tax_details['tax_amount'] += tax_values['tax_amount']
+            invoice_line_tax_details['tax_amount_currency'] += tax_values['tax_amount_currency']
+            invoice_line_tax_details['records'].append(tax_values)
```

Comment:
The logic of this whole block is the same as for the previous one (grouping by tax); can't it be factorized in a common function ?